File: frontend/src/app/core/models/alert.models.ts
export type AlertSeverity = 'Critical' | 'Warning' | 'Info';
export type AlertStatus = 'Active' | 'Acknowledged' | 'Resolved' | 'Closed';

export interface AlertResponse {
  alertEventId: number;
  severity: AlertSeverity;
  title: string;
  description: string;
  ownerTeam: string;
  status: AlertStatus;
  isActive: boolean;
  createdUtc: string;
  acknowledgedByEmail: string | null;
  acknowledgedUtc: string | null;
  resolvedByEmail: string | null;
  resolvedUtc: string | null;
  resolutionNotes: string | null;
  updatedUtc: string | null;
}

export interface CreateAlertRequest {
  severity: string;
  title: string;
  description: string;
  ownerTeam: string;
}

export interface AcknowledgeAlertRequest {
  acknowledgedByEmail: string;
}

export interface ResolveAlertRequest {
  resolvedByEmail: string;
  resolutionNotes: string | null;
}

export interface UpdateAlertOwnerRequest {
  ownerTeam: string;
}

File: backend/src/NaviFreight.Api/Models/AlertModels.cs
namespace NaviFreight.Api.Models;

public sealed record CreateAlertRequest(
    string Severity,
    string Title,
    string Description,
    string OwnerTeam);

public sealed record AcknowledgeAlertRequest(
    string AcknowledgedByEmail);

public sealed record ResolveAlertRequest(
    string  ResolvedByEmail,
    string? ResolutionNotes);

public sealed record UpdateAlertOwnerRequest(
    string OwnerTeam);
